core/src/main/scala/kafka/server/AdminManager.scala

```diff
@@ -631,7 +631,7 @@ class AdminManager(val config: KafkaConfig,
     if (configType != null)
       configType
     else
-      synonyms.iterator.map(config.typeOf).find(_ != null).orNull
+      synonyms.iterator.map(config.typeOf).find(_ != null).getOrElse(DynamicConfig.Broker.typeOf(name))
```

Comment:
Hi Brian, which config is causing issue here? Seems like `synonyms` is created from `DynamicBrokerConfig.brokerConfigSynonyms` and `brokerConfigDef` doesn't seem to contain those (at least I can't find that in code)

nit: call to `iterator` isn't needed in above line, `synonyms` is list and can be mapped directly.

Reply:
> Hi Brian, which config is causing issue here? Seems like `synonyms` is created from `DynamicBrokerConfig.brokerConfigSynonyms` and `brokerConfigDef` doesn't seem to contain those (at least I can't find that in code)
> 
> nit: call to `iterator` isn't needed in above line, `synonyms` is list and can be mapped directly.

Right, the DynamicConfig.Broker's brokerConfigDef isn't contained in DynamicBrokerConfig, and I'm not entirely sure why. I'm hoping @rajinisivaram can provide some more context here.